### PR TITLE
Tasks evaluation

### PR DESCRIPTION
**Pull request by Evil00787**
You can now evaluate tasks!
added plan details view for child' calendar

### CODE REVIEW

**Review by stasgora — CHANGES_REQUESTED** [review]

File: lib/logic/tasks_evaluation/tasks_evaluation_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		if(taskInstances.isNotEmpty) _uiTaskInstances = await _taskInstanceService.mapToUIModels(taskInstances, shouldGetTaskStatus: false);
+		else _uiTaskInstances = [];
+		_planInstanceToName = Map.fromEntries(planInstances.map((planInstance) => MapEntry(planInstance.id, nameMap[planInstance.planID])));
+
+		emit(TasksEvaluationLoadSuccess(_uiTaskInstances, _planInstanceToChild, _planInstanceToName));
+	}
+
+	void rateTask(UITaskReport report) async {
+		List<Future> updates = [];
+		int i = _uiTaskInstances.indexOf(_uiTaskInstances.singleWhere((uiTaskInstance) => uiTaskInstance.id == report.task.id));
```
[the remark this answers is not in the document]
Comment (Evil00787) [reply lib/logic/tasks_evaluation/tasks_evaluation_cubit.dart:51]: I initially wanted to update my copy of taskInstances but i realised it was pointless, nvm

File: lib/widgets/cards/report_card.dart [hunk 2]
```diff
@@ -198,9 +202,11 @@ class _ReportCardState extends State<ReportCard> {
 					subtitle: Text(
 						isRejected ?
 							AppLocales.of(context).translate('$_pageKey.raportCard.rejectedHint')
-							: AppLocales.of(context).translate('$_pageKey.raportCard.ratedOnHint', {
+							: widget.report.task.points != null ?
+							AppLocales.of(context).translate('$_pageKey.raportCard.ratedOnHint', {
                 'POINTS_NUM': (widget.report.ratingMark.value*widget.report.task.points.quantity/5.0).round().toString()
```
[the remark this answers is not in the document]
Comment (Evil00787) [reply lib/widgets/cards/report_card.dart:207]: Imo better would be just a simple function in the same file

[the remark this answers is not in the document]
Comment (stasgora) [reply lib/widgets/cards/report_card.dart]: I think you misunderstood. I was talking about only the `(widget.report.ratingMark.value*widget.report.task.points.quantity/5.0).round()` part. That's the logic (small but still). It's repeated in: `TasksEvaluationCubit`, `ReportCard` and `ReportForm`

[the remark this answers is not in the document]
Comment (stasgora) [reply lib/widgets/cards/report_card.dart]: It's actually already incorrect in both ui files as they have no `max(val, 1)` - one more reason to do this :wink:

[the remark this answers is not in the document]
Comment (Evil00787) [reply lib/widgets/cards/report_card.dart]: ahh, ok

**Review by stasgora — APPROVED** [review]

**Pull request merged**